### PR TITLE
[[ ScriptExtensions ]] Make script extensions work like lcb extensions

### PR DESCRIPTION
**Pull request by livecodeali**
To provide a bit of uniformity in the way script-library extensions
are built, packaged, and included in the IDE and standalones, this
patch rejigs some gyp files and the installer manifest to build the
script libraries into the packaged_extensions folder and include them
in the IDE Extensions folder in the installed LiveCode.

Additionally this patch:
- Tweaks the docs parser to accept title, version and author metadata
  at the library level. This info is used as the extension metadata
  in lieu of a manifest
- Adds the library level docs as metadata for the current set of
  script libraries
- Adds a script library extension packaging script, which builds
  optionally protected script libraries into the packaged_extensions
  folder

### CODE REVIEW

**Comment by livecodeali** [comment]
WIP until it resolves the 'requires' metadata issue. We should be able to use essentially the same mechanism as lcb extensions - specify the requirements as a list `name (type), ...`, topologically sort to determine load order.

File: util/package_script_extension.livecodescript [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   if the result is not 0 then
+       write "subprocess error:" && it to stderr
+       quit 2
+   end if
+end extractDocs
+ 
+command packageScriptExtension pSourceFolder, pSourceFileName, pTargetFolder, pRemoveSource   
+
+   local tTargetDir, tBuildTarget, tZip
+   put "com.livecode.script-library." & pSourceFileName into tTargetDir
```
Comment (montegoulding) [inline util/package_script_extension.livecodescript]: can you use `the short name of stack` here?

File: util/package_script_extension.livecodescript [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   put return & "zip -q -r " & tTargetDir & ".lce *" after tShell
+   put return & "popd 1>/dev/null" after tShell
+   put return & "mkdir -p " & quote & tBuildTarget & quote after tShell
+   put return & "unzip -q -o " & quote & tZip & quote && \
+         "-d " & quote & tBuildTarget & quote after tShell
+   put return & "rm" && quote & tZip & quote after tShell
+   
+   if pRemoveSource is "true" then
+      put return & "rm" && quote & tBuildTarget & slash & \ 
+         pSourceFileName & ".livecodescript" & quote after tShell
```
Comment (montegoulding) [inline util/package_script_extension.livecodescript]: I'm not sure if this makes any sense unless there's a compiled version created here (binary/password protected?)

Comment (montegoulding) [reply util/package_script_extension.livecodescript]: Scratch that I see there is!

File: Installer/package.txt [hunk 3]
```diff
@@ -450,6 +450,23 @@ component Extensions
 	into [[ToolsFolder]]/Extensions place
 		rfolder macosx:packaged_extensions/com.livecode.library.scriptitems
 
+    into [[ToolsFolder]]/Extensions place
```
Comment (montegoulding) [inline Installer/package.txt]: I think the indenting is wrong here. These files are _very_ fussy about indenting

File: util/package_script_extension.livecodescript (hunk 1 above)
Comment (livecodeali) [reply util/package_script_extension.livecodescript]: yes!

Comment (livecodeali) [reply util/package_script_extension.livecodescript]: Hmm, actually probably not yet, as the gyp has the dir hard-coded for output

**Comment by livecodeali** [comment]
Depends on https://github.com/livecode/livecode-ide/pull/1773

**Comment by livecodeali** [comment]
Here is the Travis test branch which now passes:
https://github.com/livecode/livecode/pull/6075

This build will hopefully contain all the associated changes, so needs to be checked before these PRs are merged https://vulcan.livecode.com/builders/release-commercial/builds/1665

**Comment by livecodeali** [comment]
IDE PR: https://github.com/livecode/livecode-ide/pull/1773

**Comment by montegoulding** [comment]
@livecodeali just incase you have't seen it there's conflicts to resolve here

**Comment by livecodeali** [comment]
Gah, again! I just resolved some

**Comment by montegoulding** [comment]
It's pretty annoying that github doesn't send the author an email about conflicts...

**Comment by montegoulding** [comment]
@livecode-vulcan review ok 006944d

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok 006944d80e891ad5a1818ef9bd9c04c0dbae1b3c

**Comment by livecode-vulcan** [comment]
😎 test success 006944d80e891ad5a1818ef9bd9c04c0dbae1b3c

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/969)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/985)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/959)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/32)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/32)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/963)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/986)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/969)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/985)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/968)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/978)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1000)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/964)

**Pull request merged**